### DISCUSSION TITLE
Delay to switch themes

### DISCUSSION THREAD

**Discussion by RiczCalixto** [jpudysz/react-native-unistyles#212] [Q&A] [answered]
Hello, how are you?

I am using UnistylesRuntime to switch themes, but I noticed there is a small delay when I execute the setTheme function.

I followed the documentation, but I couldn't identify if I am doing something wrong or if this delay is actually expected.

I recorded a video that demonstrates the behavior. It doesn't appears, but I press to change theme when start to record.. So it take 5s to change theme.

Thank you for your time and attention.

https://github.com/jpudysz/react-native-unistyles/assets/38529592/05da935c-513b-49e9-82d5-d26529b7f1c7

![image](https://github.com/jpudysz/react-native-unistyles/assets/38529592/c1019db4-a9b8-46f1-9e1e-fddd9f86c2d9)
![image](https://github.com/jpudysz/react-native-unistyles/assets/38529592/481df28e-46d1-4173-a53b-e2b7eba92252)
![image](https://github.com/jpudysz/react-native-unistyles/assets/38529592/4bf244d2-1fc5-402b-9f1f-d565666028c3)

**Comment by jpudysz** [comment, marked as answer]
It just one big re-render of your app. If eg. your root component has useStyles, entire app will re-render. Same goes if you have mounted FlatList with hundreds of components and each consist of next components with useStyles. 

This is how React works 🤷‍♂️ try to limit the number of mounted hooks while switching the themes, otherwise it will look like this. 

I was debugging such case on discord for someone and it appeared that this dev had mounted FlatList with ~80 components each with ~12 hooks, so app was re-rendering 1000 components while switching themes. That's only FlatList.. and what about headers, tab bars and other screens.
